check/src/kindcheck.rs

```diff
@@ -171,6 +171,14 @@ impl<'a> KindCheck<'a> {
             }
             Type::Variable(_) => Ok((self.subs.new_var(), typ.clone())),
             Type::Builtin(builtin_typ) => Ok((self.builtin_kind(builtin_typ), typ.clone())),
+            Type::Forall(ref params, ref typ) => {
+                let (ret_kind, ret_typ) = self.kindcheck(typ)?;
+                let kind = params.into_iter().rev().fold(ret_kind, |body, param| {
+                    Kind::function(param.kind.clone(), body)
+                });
+
+                Ok((kind, ret_typ.clone()))
+            }
```

Comment:
I don't think this is right. When encountering a `forall` the kindchecking needs to introduce variables for `params` and the resulting kind ought to be the kind of `typ`.

Does that make sense?